lib/fog/aws/requests/rds/modify_db_instance.rb

```diff
@@ -61,6 +61,9 @@ def modify_db_instance(db_name, apply_immediately, options={})
               #else
               #  modified_server = server["PendingModifiedValues"].merge!(options) # it appends
               #end
+              if options["NewDBInstanceIdentifier"]
+                options["DBInstanceIdentifier"] = options.delete("NewDBInstanceIdentifier")
```

Comment:
That is not the API behavior.  This attribute takes effect half instantly.  The ID you query is the old id, but the ID returned is the new one.  This never goes into `PendingModifiedValues`